### PR TITLE
Update rufus internal partition state after changing drive partitions

### PR DESCRIPTION
**Pull request by ManMower**
Many of rufus' internal functions (AltMountVolume, GetLogicalName, etc)
use information it caches. When we change the partition table behind it's
back, it may no longer function properly.

This is easily reproduced by writing a USB stick with 2 partitions, by
making an endless live USB with dd, then trying to overwrite it with rufus.
The new partition scheme has 3 partitions, and the core rufus functions
can't find the volume name for the 3rd partition because it didn't exist
last time GetPartitionData() was called.

We fix this by calling GetPartitionData() after repartitioning to update
internal state.

https://phabricator.endlessm.com/T29676

### CODE REVIEW

File: endless/src/endless/EndlessUsbToolDlg.cpp [hunk 1]
```diff
@@ -4883,6 +4884,14 @@ DWORD WINAPI CEndlessUsbToolDlg::CreateUSBStick(LPVOID param)
 
 	CHECK_IF_CANCELLED;
 
+	/* We changed the partition table, so we need to get rufus to update its internal state so
+	 * stuff like FormatPartition and AltMountVolume can figure out volume names.
+	 * GetDrivePartitionData gets the physical lock itself, so we have to drop that first. */
+	safe_unlockclose(hPhysical);
+	GetDrivePartitionData(SelectedDrive.DeviceNumber, tmp_fs_name, sizeof(tmp_fs_name), FALSE);
+	hPhysical = GetPhysicalHandle(DriveIndex, TRUE, TRUE, FALSE);
+	IFFALSE_GOTOERROR(hPhysical != INVALID_HANDLE_VALUE, "Error on acquiring disk handle.");
```
Comment (wjt) [inline endless/src/endless/EndlessUsbToolDlg.cpp]:
```suggestion
	IFFALSE_GOTOERROR(hPhysical != INVALID_HANDLE_VALUE, "Error on reacquiring disk handle.");
```

**Review by wjt — APPROVED** [review]
Argh, so this would lead to a situation where overwriting a USB stick which had previously been formatted with our layout would work, but any other layout would fail? :'(

**Comment by ManMower** [comment]
> Argh, so this would lead to a situation where overwriting a USB stick which had previously been formatted with our layout would work, but any other layout would fail? :'(

Exactly - so since I was testing by overwriting the same stick many times, I probably saw it once a long time ago and thought I "fixed" it...

Obviously this is a case the test plan needs to exercise.

This is definitely something I introduced during the rebase. I've tested and it doesn't happen on the currently released installer on the endless web page.

I haven't really done any investigation to figure out exactly why.

File: endless/src/endless/EndlessUsbToolDlg.cpp (hunk 1 above)
Comment (ManMower) [reply endless/src/endless/EndlessUsbToolDlg.cpp]: Done

**Comment by wjt** [comment]
> This is definitely something I introduced during the rebase. I've tested and it doesn't happen on the currently released installer on the endless web page.

Well, the reason I asked is that I definitely remember seeing symptoms like this 3 years ago but never could work it out

**Pull request merged**